x-pack/legacy/plugins/apm/public/components/app/Settings/UiIndices/index.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiButton
+} from '@elastic/eui';
+import { useFetcher } from '../../../../hooks/useFetcher';
+import { useCallApmApi } from '../../../../hooks/useCallApmApi';
+import { APMClient } from '../../../../services/rest/createCallApmApi';
+
+const INDICES = [
+  { configuration: 'apm_oss.sourcemapIndices', label: 'Sourcemap Indices' },
+  { configuration: 'apm_oss.errorIndices', label: 'Error Indices' },
+  { configuration: 'apm_oss.onboardingIndices', label: 'Onboarding Indices' },
+  { configuration: 'apm_oss.spanIndices', label: 'Span Indices' },
+  { configuration: 'apm_oss.transactionIndices', label: 'Transaction Indices' },
+  { configuration: 'apm_oss.metricsIndices', label: 'Metrics Indices' },
+  {
+    configuration: 'apm_oss.apmAgentConfigurationIndex',
+    label: 'Agent Configuration Index'
+  }
+];
+
+async function saveUiIndices({
+  callApmApi,
+  uiIndices
+}: {
+  callApmApi: APMClient;
+  uiIndices: { [key: string]: string };
+}) {
+  await callApmApi({
+    method: 'POST',
+    pathname: '/api/apm/settings/ui-indices/save',
+    params: {
+      body: uiIndices
+    }
+  });
+}
+
+export function UiIndices() {
+  const [uiIndices, setUiIndices] = useState<{ [key: string]: string }>({});
+  const [isSaving, setIsSaving] = useState(false);
+
+  const callApmApiFromHook = useCallApmApi();
```

Comment:
Is it necessary to say it comes from a hook? I got a little confused with that name at first.

```suggestion
  const callApmApi = useCallApmApi();
```

Reply:
Why not use `calApmApi` from the hook everywhere?

```ts
  const callApmApi = useCallApmApi();

  const { data = [], status, refetch } = useFetcher(
   () => callApmApi({ pathname: `/api/apm/settings/ui-indices` }),
    []
  );
```

I think it's confusing having two different instances of it.